src/components/fullscreen/FullscreenContext.tsx

```diff
@@ -43,20 +43,34 @@ export function FullScreenProvider(props: FullscreenProps) {
 }
 function FullscreenInner(props: FullscreenProps) {
   const { children } = props;
-  const { isFullScreen } = useFullscreen();
+  const { isFullScreen, toggle } = useFullscreen();
   const ref = useRef<HTMLDivElement>(null);
   useEffect(() => {
-    if (isFullScreen && ref.current) {
-      ref.current.requestFullscreen().catch(() => {
+    function onFullscreenChange() {
+      const value = Boolean(document.fullscreenElement);
+      if (!value && isFullScreen) toggle();
+    }
+    const div = ref.current;
+    div?.addEventListener('fullscreenchange', onFullscreenChange);
+    div?.addEventListener('mozfullscreenchange', onFullscreenChange);
+    div?.addEventListener('MSFullscreenChange', onFullscreenChange);
+    div?.addEventListener('webkitfullscreenchange', onFullscreenChange);
```

Comment:
You can keep only `fullscreenchange`: https://caniuse.com/mdn-api_element_fullscreenchange_event
We don't care about IE11 and old browsers.

Reply:
fixed